### ISSUE TITLE
Boolean HTML attributes not polyglot 

### ISSUE THREAD

**Issue by marrus-sh** [jgm/pandoc#5780]
Pandoc version: 2.7.3

I am compiling Markdown with pandoc to standalone HTML, using a custom template of my own with the `--template` flag. The full call is something along the lines of:

    pandoc -f markdown-smart -t html5-smart --standalone \
        --template "template.xhtml" --filter pandoc-citeproc \
        --self-contained --section-divs --mathml

In the template I have code like the following:

```html
<div hidden="">
```

However, the HTML which results is instead:

```html
<div hidden>
```

(The `hidden` attribute has lost its `=""`.)

This is equivalent HTML, but it is not valid polyglot markup (it is invalid XML). One can get around this by instead writing:

```html
<div hidden="hidden">
```

(Which is defined by HTML5 to be equivalent.) But it would be nice if pandoc supported polyglot empty attributes as well.

**Comment by marrus-sh** [comment]
Doing some additional testing, it seems like `--self-contained` specifically is the culprit here; correct polyglot markup is produced when I remove that option.

**Comment by mb21** [comment]
We're using the tagsoup library's data model to implement the `--self-contained` option. Essentially we re-parse the string generated by the HTML writer using tagsoup, replace all the external things with data-urls, then render with tagsoup to HTML again. See [SelfContained.hs](https://github.com/jgm/pandoc/blob/master/src/Text/Pandoc/SelfContained.hs#L261).

Seems like they have an old open issue about this: https://github.com/ndmitchell/tagsoup/issues/23 and there's currently no workaround?

**Comment by jgm** [comment]
Does `<div hidden="">` have the same meaning in HTML as `<div hidden="hidden">`?
If so, we could replace the first with the second automatically before calling tagsoup.

**Comment by mb21** [comment]
> Does <div hidden=""> have the same meaning in HTML as <div hidden="hidden">?

yes, https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes#Boolean_Attributes

but I guess it would be better if it were fixed in tagsoup directly...
